honeybee_vtk/config.py

```diff
@@ -208,114 +231,84 @@ def _validate_data(data: DataConfig, model: Model) -> bool:
     Returns:
         A boolean value.
     """
-    # if file_paths is empty
-    if not data.file_paths:
-        raise ValueError(
-            f'For object with name {data.name} There are not file paths'
-            ' provided to load data from.'
+    grids_info_json = pathlib.Path(data.path).joinpath('grids_info.json')
+    # TODO: Confirm with Chris if this check for json is needed. I think it's not.
+    try:
+        with open(grids_info_json) as fh:
+            grids_info = json.load(fh)
+    except json.decoder.JSONDecodeError:
+        raise TypeError(
+            'grids_info is not a valid json file.'
         )
-
-    # if object name is "grid" check that the name of files match the grid names
-    if data.object_type == model_dataset_names[-1]:
-
-        # make sure grids are loaded on the model if grid data is to be mounted
+    else:
         assert len(model.sensor_grids.data) > 0, 'Sensor grids are not loaded on'\
             ' this model. Reload them using grid options.'
-
-        grid_names = [grid.identifier for grid in model.sensor_grids.data]
-        file_names = [pathlib.Path(path).stem for path in data.file_paths]
-        names_in_grids = all([name in grid_names for name in file_names])
-
-        # the number of data files must not exceed the number of grids
-        if len(file_names) > len(grid_names):
-            raise ValueError(
-                f'There are only {len(grid_names)} grids in the model and the'
-                f' identifiers of those grids are {tuple(grid_names)}.'
-            )
-
-        # TODO: remove this check in the next iteration
-        # make sure there's one file for each grid in the model
-        if not names_in_grids:
-            raise ValueError(
-                'Make sure the file names match the grid identifiers. The'
-                f' identifiers of the grids in the model are {tuple(grid_names)}.'
-            )
+        # TODO: Make sure to remove this limitation. A user should not have to always
+        # TODO: load all the grids
+        assert len(model.sensor_grids.data) == len(grids_info), 'The number of result'\
+            f' files {len(grids_info)} does for {data.identifier} does not match'\
+            f' the number of sensor grids in the model {len(model.sensor_grids.data)}.'
+
+        grids_model_identifiers = [grid.identifier for grid in model.sensor_grids.data]
+        grids_info_identifiers = [grid['identifier'] for grid in grids_info]
```

Comment:
You should actually use the `full_identifier` property in the girds_info and in the model sensorgrid to match the result file to the correct grid in the model.

Reply:
Complied